packages/otelbin/src/components/monaco-editor/parseYaml.ts

```diff
@@ -70,14 +76,30 @@ export interface IValidateItem {
 	[key: string]: ILeaf[];
 }
 
-export const getParsedValue = (editorValue: string) => {
+export const getYamlDocument = (editorValue: string) => {
 	const value = editorValue;
 	const parsedYaml = Array.from(new Parser().parse(value));
 	const doc = parsedYaml.find((token) => token.type === "document") as Document;
 	const docElements: IItem[] = doc?.value?.items ?? [];
 	return docElements;
 };
 
+export function parseYaml(yamlItems: IItem[]) {
```

Comment:
parseYaml function is more efficient and accessible function to make the big yaml Document smaller, through the cleanups, will adjust other parsing functions with this one. @bripkens